### PR TITLE
New welcome email

### PR DESCRIPTION
**Pull request by hteumeuleu**
This PR updates the welcome email to its new design final form. 

[Screenshot of the email](https://github.com/mastodon/mastodon/assets/3451753/f4c62a5f-e233-4b5b-b2f4-6737c2937c45)

It's still incomplete though and there are a lots of points I would need help on. 

1. Where should the "Sign in" button go in the header part?
2. In the welcome checklist, the original design had parts that could turn green if a step was completed. I've implemented this by passing a `checked` variable to the `_checklist` partial. But are those even possible to track, dev-wise? (If so, how?)
3. In the welcome checklist, where should each action button point to?
4. The apps buttons are images with text on them. The `alt` is managed via translation files. But how can we manage the images for different languages?
5. The follow and hashtags list are absent for now (but the partials are ready). Gargron shared this code ([1]) for this but I'm don't know Ruby enough to be able to do this myself.
6. What should the "View more people to follow" and "view more trending hashtags" links point to?
7. What should each of the four "Learn more" links in the features section point to?
8. `haml-lint` complains about lines too long but I'm not sure how to fix this.

[1] Code shared by gargron to get recommended accounts and trending hashtags
```
who_to_follow = AccountSuggestions.new(current_account).get(5)
trending = Trends.tags.query.allowed.limit(5)
```

___

Fixes MAS-147

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
> It's still incomplete though and there are a lots of points I would need help on.
> 
>     1. Where should the "Sign in" button go in the header part?

`new_user_session_url` should work!

>     2. In the welcome checklist, the original design had parts that could turn green if a step was completed. I've implemented this by passing a `checked` variable to the `_checklist` partial. But are those even possible to track, dev-wise? (If so, how?)

That is a good question… while we could check at the time the email is sent, the email is sent pretty much immediately, so in practice those are unlikely to ever be checked.

>     3. In the welcome checklist, where should each action button point to?

That is also a good question.
- personalize: either `settings_profile_url` or `web_url('start/profile')`
- follow: `web_url('start/follows')`
- compose: this one is tricky… `web_url('publish')` would make sense on mobile but not on a larger screen. Maybe just `web_url` (the whole web app) or `share_url` (just a compose view, outside of the main web app), but none feels satisfactory
- `share`: no idea, the closest I can think of is the user's profile… `short_account_url(@resource.account)`

>     4. The apps buttons are images with text on them. The `alt` is managed via translation files. But how can we manage the images for different languages?

I don't know. On the subject, while I suppose we are allowed to redistribute those images, I assume this might be under a different license than the rest of Mastodon?

>     5. The follow and hashtags list are absent for now (but the partials are ready). Gargron shared this code ([1]) for this but I'm don't know Ruby enough to be able to do this myself.

Alright, I can take a look later.

>     6. What should the "View more people to follow" and "view more trending hashtags" links point to?

For “View more people to follow” I'd say either `web_url('start/follows')` or `web_url('explore/suggestions')`. For trending hashtags, this would be `web_url('explore/tags')`.

>     7. What should each of the four "Learn more" links in the features section point to?

The same as the corresponding links on https://joinmastodon.org/

>     8. `haml-lint` complains about lines too long but I'm not sure how to fix this.

I will have a look and add an inline suggestion when I have one!

**Comment by ClearlyClaire** [comment]
> > ```
> > 8. `haml-lint` complains about lines too long but I'm not sure how to fix this.
> > ```
> 
> 
> I will have a look and add an inline suggestion when I have one!

I only see it complaining about the whole template length, not line length. The mail is conceptually split in 3 pieces, those could be three different partials. Alternatively, you could ignore this for now.

**Comment by mjankowski** [comment]
@hteumeuleu would it help to have another branch on my fork for this one that you could pull commits off of? I could attempt to fill in some of the backend stuff and handle linting issues similar to earlier PR.

**Comment by Gargron** [comment]
Template should be ready now. I've added a one hour delay to sending the welcome e-mail so the user has some time to go through in-app onboarding first, otherwise the checklist would always have all items unchecked.

File: .haml-lint.yml
Comment (mjankowski) [inline .haml-lint.yml:16]: This is not the worst thing ever, but we were making good progress on this in the other direction.

Could we convert lines 28-32 and 73-76 in welcome.html.haml into more diffable multi-line arg style and/or add helper methods for "mailer checklist item" and "mailer feature"

File: app/javascript/images/mailer-new/welcome/feat1.png
Comment (mjankowski) [inline app/javascript/images/mailer-new/welcome/feat1.png:1]: Minor thing, but -- could we name these files to align with the i18n values for each feature? (`feature_control`, ...)

File: app/views/user_mailer/welcome.html.haml
Comment (mjankowski) [inline app/views/user_mailer/welcome.html.haml:73]: Same thing here as the checklist -- could prob use a presenter class or a hash in the mailer to define the features and iterate through them.

File: .haml-lint.yml
Comment (Gargron) [reply .haml-lint.yml:16]: Ah, but I converted them back from multi-line because the linter was complaining about the line count in the view.

Comment (mjankowski) [reply .haml-lint.yml:16]: Ah - yeah when fully unrolled I guess it puts the whole file over that other limit...

Could either:

- Merge as-is, bumping this limit, maybe deal w/ it in refactor later on
- Pull out "checklists" and "features" partials which keep the hard-coding, but go back to multi-line (would keep welcome.html under 100, and each partial would also - I think? - be under that)
- Do something with presenter classes or other structure so we're doing a collection render in welcome.html and the looped objects know how to supply the partial with each attribute

**Comment by brendanjones** [comment]
This is more a note for the future, but IMO it'd be good to replace the link to trending hashtags with a link to a curated onboarding flow for hashtags, like described in https://github.com/mastodon/mastodon/issues/20854#issuecomment-1947321585.

Trending hashtags is often full of weird, indecipherable or just plain boring (e.g. #Monday) tags, so I don't think I've ever followed a hashtag from the trending hashtags list. It's more a curiosity, and I don't think sending new users there is very valuable.

The email looks fantastic, amazing work.

File: app/views/application/mailer/_feature.html.haml
Comment (mjankowski) [inline app/views/application/mailer/_feature.html.haml:14]: Are there scenarios where these local vars are not defined - or is this just overly defensive check?

File: app/views/user_mailer/welcome.html.haml
Comment (mjankowski) [inline app/views/user_mailer/welcome.html.haml:73]: If we wanted to go one further here -- the `feature_title` and `feature_text` values are derivable from just the `key` -- we could pass in just the key and the final bool value to the partial and let it interpolate the i18n values.

Comment (Gargron) [reply app/views/user_mailer/welcome.html.haml:73]: I was considering this but then i18n-tasks wouldn’t be able to keep track of these strings so I’m not sure if it’s worth it.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by mjankowski** [comment]
This looks great - glad it's in. Might open some tidy-up PRs on the mailer views/partials.

One tiny thing I noticed in preview -- looks like when we converted the button text from placeholder to actual labels, there's a width/wrapping issue (see "personalize" button in image)...

Screenshot 2024-03-01 at 12 01 03 PM

**Comment by colinstu12** [comment]
@Gargron How does this 1 hour delay work exactly? 

Frustratingly for instances where approval is required, users will not see any onboarding or approval email until an hour after the approval has been made. This is a lot of time to leave the user wondering what is happening (unless they attempt to login again right away). 

Could this delay be made configurable? And/or only trigger it when manual approval is required? Such a delay makes sense if an invite link was used and the user is logged into their approved account straight away. 

edit: I see it was added here https://github.com/mastodon/mastodon/commit/934cab75083a4ab79003a583881775b6d4a2b58a#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29b

**Pull request merged**
